spec.emu

```diff
@@ -4,20 +4,2216 @@
+      <h1>Static Semantics: LexicallyScopedDeclarations ( ): a List of Parse Nodes</h1>
+      <dl class="header">
+      </dl>
+      <emu-grammar>
+        <ins>ModuleItem : ModuleDeclaration</ins>
+      </emu-grammar>
+      <emu-alg>
+        1. Return a List whose sole element is |ModuleDeclaration|.
+      </emu-alg>
+      <emu-grammar>
+        <ins>ScriptItem : ModuleDeclaration</ins>
+      </emu-grammar>
+      <emu-alg>
+        1. Return a List whose sole element is |ModuleDeclaration|.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-containsundefinedmodulereference" type="sdo" number="12">
+      <h1>
+        <ins>
+          Static Semantics: ContainsUndefinedModuleReference (
+            _declaredModules_: a List of Strings,
+          ): a Boolean
+        </ins>
+      </h1>
+      <dl class="header">
+      </dl>
+      <p>Unless explicitly specified otherwise, all nonterminals have an implicit definition for ContainsUndefinedModuleReference with argument _declaredModules_.The implicit definition applies ContainsUndefinedModuleReference with argument _declaredModules_ to the nonterminal's inner nonterminals, and returns *true* if it's *true* for any of them. For example, ContainsUndefinedModuleReference of |ExponentiationExpression| is implicitly defined as follows:</p>
+      <emu-grammar example>
+        ExponentiationExpression : UpdateExpression `**` ExponentiationExpression
+      </emu-grammar>
+      <emu-alg example>
+        1. Let _hasUndefinedRef_ be ContainsUndefinedModuleReference of |UpdateExpression| with argument _declaredModules_.
+        1. If _hasUndefinedRef_ is *true*, return *true*.
+        1. Return ContainsUndefinedModuleReference of |ExponentiationExpression| with argument _declaredModules_.
+      </emu-alg>
+
+      <p>The following productions have a different definition of ContainsUndefinedModuleReference:</p>
+
+      <emu-grammar>
+        ModuleSpecifier : Identifier
+      </emu-grammar>
+      <emu-alg>
+        1. Let _reference_ be the StringValue of |Identifier|.
+        1. If _reference_ is in _declaredModules_, return *false*.
+        1. Return *true*.
+      </emu-alg>
+
+      <emu-grammar>
+        ModuleBody : ModuleItemList
+      </emu-grammar>
+      <emu-alg>
+        1. Let _localDeclaredModules_ be DeclaredModuleNames of |ModuleItemList|.
+        1. Let _lexNames_ be the LexicallyDeclaredNames of |ModuleItemList|.
+        1. Let _varNames_ be the VarDeclaredNames of |ModuleItemList|.
+        1. Let _visibleDeclaredModules_ be a new List containing the elements of _declaredModules_ that are not in _lexNames_ or _varNames_.
+        1. Append all the elements of _localDeclaredModules_ to _visibleDeclaredModules_.
+        1. Return ContainsUndefinedModuleReference of |ModuleItemList| with argument _visibleDeclaredModules_.
+      </emu-alg>
+
+      <emu-grammar>
+        ScriptBody : ScriptItemList
```

Comment:
Yes, `A` becomes a global reference because that code behaves like this:
```html
<script>
const A = module {}
</script>
<script>
import(A);
</script>
```